src/Encoder.ts

```diff
@@ -223,18 +224,12 @@ export class Encoder {
     }
   }
 
-  countObjectKeys(object: Record<string, unknown>): number {
-    let count = 0;
-    for (const key in object) {
-      if (Object.prototype.hasOwnProperty.call(object, key)) {
-        count++;
-      }
-    }
-    return count;
-  }
-
   encodeMap(object: Record<string, unknown>, depth: number) {
-    const size = this.countObjectKeys(object);
+    const keys = Object.keys(object);
+    if (this.sortKeys) {
```

Comment:
Maybe we should have two separate branches? One with keys array and one without? it will be better from performance perspective. But maybe it's over optimization

Reply:
Maybe you are right because most of the cases do not set `sortKeys`.

Reply:
Unfortunately, `Object.keys()` version is always faster.

benchmark script:

```typescript
import { encode, decode } from "../src";
// @ts-ignore
import _ from "lodash";

const data: Record<string, number> = {};
for (let i = 0; i < 100; i++) {
  data["foo" + i] = i;
}

// warm up
const encoded = encode(data);
decode(encoded);

// run

console.time("encode sortKeys: false");
for (let i = 0; i < 100000; i++) {
  encode(data, { sortKeys: false });
}
console.timeEnd("encode sortKeys: false");

console.time("encode sortKeys: true");
for (let i = 0; i < 100000; i++) {
  encode(data, { sortKeys: true });
}
console.timeEnd("encode sortKeys: true");
```

result of `for-in` version:

```
$ npx ts-node benchmark/sandbox.ts
encode sortKeys: false: 2681.488ms
encode sortKeys: true: 2374.899ms
```

result of `Object.keys()` version (i.e. the same as this PR):

```
$ npx ts-node benchmark/sandbox.ts
encode sortKeys: false: 1336.151ms
encode sortKeys: true: 2343.010ms
```

I think `for-in` version uses less memory, but it's much slower.

So I don't change this PR.